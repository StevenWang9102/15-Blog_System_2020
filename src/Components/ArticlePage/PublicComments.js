import React from "react";
import dateFormat from "dateformat";

export const PublicComments = props => {
  return (
    <div className='row'>
      <div className='col-xs-12 col-md-8 offset-md-2'>
        {props.currentComments &&
          props.currentComments.comments &&
          props.currentComments.comments.map((comment, index) => {
            return (
              <div key={index}>
                <div className='card'>
                  <div className='card-block'>
                    <p className='card-text'>{comment.body}</p>
                  </div>

                  <div className='card-footer'>
                    <a href='#top' className='comment-author'>
                      <img
                        src={comment.author.image}
                        className='comment-author-img'
                        alt=''
                      />
                    </a> &nbsp;

                    <a href='#top' className='comment-author'>
                      {comment.author.username}
                    </a>

                    <span className='date-posted'>
                      {dateFormat(comment.updatedAt, "ddd mmm dd yyyy")}
                    </span>

                  </div>
                </div>
              </div>
            );
          })}
      </div>
    </div>
  );
};
