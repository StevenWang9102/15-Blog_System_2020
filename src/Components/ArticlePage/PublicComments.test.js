import React from "react";
import { PublicComments } from "./PublicComments";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <PublicComments
        currentComments={{}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
