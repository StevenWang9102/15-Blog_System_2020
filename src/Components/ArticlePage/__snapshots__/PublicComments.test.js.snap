// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 col-md-8 offset-md-2"
  />
</div>
`;
